### PR TITLE
#F82ebvCI-ch-setup-testing-environment with-jest-and-enzyme ,and Travis -Ci

### PR DESCRIPTION
**Pull request by Mugema123**
**What does this PR do?**
`Setup testing environment with jest and enzyme`
`Travis CI is configured as a continuous integration service used to build and test the app.`
**Description of Task to be completed?**
`Configured Testing environment with Jest & Enzyme to ensure that all functionalities are tested before deployment.`
`Travis CI is configured as a continuous integration service used to build and test the app.`
`Travis CI badge has been added to your repo`
`Configure CodeClimate for code analysis and Maintainability checking.`

**How should this be manually tested?**
`Clone the repo locally`
`run npm i`
`run npm test`

**What are the relevant pivotal tracker stories?**
#F82ebvCI
**Screenshots (if appropriate)**
m1
m2

### CODE REVIEW

**Review by Ntwali-Josue — COMMENTED** [review]
try to not push node_modules and .vscode folders

**Review by Ntwali-Josue — DISMISSED** [review]

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "git.ignoreLimitWarning": true
+}
```
Comment (Ntwali-Josue) [inline .vscode/settings.json]: please add this vscode file in your gitignore

File: .babelrc [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+    "presets": ["@babel/preset-env", "@babel/preset-react"],
+    "plugins": [
+      ["@babel/transform-runtime"]
+  ]
+
+  }
+
+
```
Comment (Ntwali-Josue) [inline .babelrc]: At the end of each file of yours add a single empty line.

**Review by Ally4 — DISMISSED** [review]

**Review by TuyizeeAnastase — COMMENTED** [review]
can you use first letter capitalized
 for components.

**Review by TuyizeeAnastase — DISMISSED** [review]

**Review by Ally4 — DISMISSED** [review]

**Review by Ntwali-Josue — DISMISSED** [review]

File: package.json [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    "jest": "^26.6.3",
+    "jest-enzyme": "^7.1.2",
+    "react": "^17.0.1",
+    "react-dom": "^17.0.1",
+    "webpack-cli": "^4.2.0",
+    "webpack-dev-server": "^3.11.0"
+  },
+  "dependencies": {
+    "@babel/plugin-transform-runtime": "^7.12.1",
+    "@webpack-cli/init": "^1.0.3",
```
Comment (ceelogre) [inline package.json]: Do we really need to have these in dependencies?

File: package.json [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    "runtime": "^0.14.1",
```
Comment (ceelogre) [inline package.json]: What's the use of this package?

File: src/App.test.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+
+import App from './app'
+
+describe('App', () => {
+  test('should test initial message', () => {
+    const wrapper = shallow(
+      <App />
+    )
+
```
Comment (ceelogre) [inline src/App.test.js]: There's no 'initial message' tested here.

**Review by ceelogre — COMMENTED** [review]
Look at the comments and rebase so we can proceed.

**Review by Ntwali-Josue — DISMISSED** [review]

**Review by TuyizeeAnastase — DISMISSED** [review]

**Comment by Ally4** [comment]
Hi Ronald, First change the label on this PR, and work on the coverage, and change the screenshot in the PR, otherwise it a good job

**Comment by Ally4** [comment]
give also the badges their own field, not in the installation field

**Comment by Ally4** [comment]
give also the badges their own field, not in the installation field

File: coverage/lcov-report/index.html [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+                <div class='push'></div><!-- for sticky footer -->
+            </div><!-- /wrapper -->
+            <div class='footer quiet pad2 space-top1 center small'>
+                Code coverage generated by
+                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
+                at Wed Nov 18 2020 13:59:19 GMT+0200 (Central Africa Time)
+            </div>
+        </div>
+        <script src="prettify.js"></script>
+        <script>
```
Comment (Ntwali-Josue) [inline coverage/lcov-report/index.html]: don't push coverage

File: coverage/lcov-report/sorter.js [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+                return -1 * sorter(a, b);
+            };
+        }
+
+        for (i = 0; i < rowNodes.length; i += 1) {
+            rows.push(rowNodes[i]);
+            tableBody.removeChild(rowNodes[i]);
+        }
+
+        rows.sort(finalSorter);
```
Comment (Ntwali-Josue) [inline coverage/lcov-report/sorter.js]: remove these coverage files

**Review by Ntwali-Josue — CHANGES_REQUESTED** [review]

**Review by Ntwali-Josue — DISMISSED** [review]

**Review by TuyizeeAnastase — DISMISSED** [review]

**Review by Ally4 — DISMISSED** [review]

**Review by Ntwali-Josue — APPROVED** [review]

**Review by ceelogre — APPROVED** [review]

**Pull request merged**